Library/Homebrew/livecheck/strategy/cpan.rb

```diff
@@ -0,0 +1,67 @@
+# typed: false
+# frozen_string_literal: true
+
+require "uri"
+
+module Homebrew
+  module Livecheck
+    module Strategy
+      # The {Cpan} strategy identifies versions of software at
+      # cpan.metacpan.org by checking directory listing pages.
+      #
+      # CPAN URLs take the following format:
+      #
+      # * `https://cpan.metacpan.org/authors/id/M/MI/MIYAGAWA/Carton-v1.0.34.tar.gz`
```

Comment:
```suggestion
      # * `https://cpan.metacpan.org/authors/id/H/HO/HOMEBREW/Brew-v1.2.3.tar.gz`
      # * `https://cpan.metacpan.org/authors/id/H/HO/HOMEBREW/brew/brew-v1.2.3.tar.gz`
      #
      # In these examples, `HOMEBREW` is the author name and the preceding `H`
      # and `HO` directories correspond to the first letter(s). Some authors
      # also store files in subdirectories, as in the second example above.
```

I usually make the example URLs in the documentation into something generic that doesn't exist but still demonstrates the format of the URL in an understandable way. It's not technically required but I think it makes the documentation predictable and also avoids the issue of someone coming our way and asking to not be mentioned in the docs.

Reply:
I would argue that documentation should always be verifiable and having generic/non-existent URLs makes it really annoying but at the same time, you are right about _unwanted promotion_.  I don't know, I belive in your judgment 🤝